tests/typecheck-tests.ts

```diff
@@ -551,6 +552,71 @@ import { ok, err, okAsync, errAsync, fromSafePromise, Result, ResultAsync } from
 
 
 (function describe(_ = 'Combine on Unbounded lists') {
-  // TODO:
-  // https://github.com/supermacro/neverthrow/issues/226
+  (function describe(_ = 'combine') {
+    (function it(_ = 'combines different results into one') {
+      type Expectation = Result<[ number, string, boolean, boolean ], Error | string | string[]>;
+
+      const result: Expectation = Result.combine([
+        ok<number, string>(1),
+        ok<string, string>('string'),
+        err<boolean, string[]>([ 'string', 'string2' ]),
+        err<boolean, Error>(new Error('error content')),
+      ])
+    });
+
+    (function it(_ = 'combines only ok results into one') {
+      type Expectation = Result<[ number, string ], never>;
+
+      const result: Expectation = Result.combine([
+        ok(1),
+        ok('string'),
+      ]);
+    });
+
+    (function it(_ = 'combines only err results into one') {
+      type Expectation = Result<[ never, never ], number | string>;
+
+      const result: Expectation = Result.combine([
+        err(1),
+        err('string'),
+      ]);
+    });
+
+    (function it(_ = 'combines empty list results into one') {
+      type Expectation = Result<never[], never>;
+
+      const result: Expectation = Result.combine([]);
+    });
+  });
+
+  (function describe(_ = 'combineWithAllErrors') {
+    (function it(_ = 'combines different results into one') {
+      type Expectation = Result<[ number, string, never, never ], [never, never, string[], Error]>;
+
+      const result: Expectation = Result.combineWithAllErrors([
+        ok(1),
+        ok('string'),
+        err([ 'string', 'string2' ]),
+        err(new Error('error content')),
+      ]);
+    });
+
+    (function it(_ = 'combines only ok results into one') {
+      type Expectation = Result<[ number, string ], never[]>;
+
+      const result: Expectation = Result.combineWithAllErrors([
+        ok(1),
+        ok('string'),
+      ]);
+    });
+
+    (function it(_ = 'combines only err results into one') {
+      type Expectation = Result<never, [number, string]>;
```

Comment:
```suggestion
      type Expectation = Result<never[], [number, string]>;
```

This is odd, it seems like there is different behaviour when passing similar arguments. Compare this expectation to this one:

https://github.com/supermacro/neverthrow/pull/418/files#diff-5e6df76f1ff5c2ddfbb0bee2e515236794778a429792a91d3b853bfecc884c16R577

Reply:
@supermacro  These two are different, one is a tuple where the other is an array. The tuple has fixed size where the array does not. I have preferred to use tuple as it would be providing a better type and length info.